### ISSUE TITLE
How does web application based on asyncio use minio-py?

### ISSUE THREAD

**Issue by Tioit-Wang** [minio/minio-py#1229]
How does web application based on asyncio use minio-py?

I don't think any asynchronous program can stand being blocked by synchronous api for 300ms on api involving minio-py.

Multi-thread is not friendly enough for asynchrony.

Now there are more and more web programs based on asyncio. I think it is necessary to support asyncio, and it can keep minio competitive in the field. After all, ecology is very important.

**Comment by Tioit-Wang** [comment]
As [#536](https://github.com/minio/minio-py/pull/536) said

**Comment by balamurugana** [comment]
`urllib3` doesn't support async, hence `minio-py`. You could write your own layer on top of `minio-py`.

**Comment by Trolldemorted** [comment]
Your dependencies' problems are your problems - is switching to aiohttp or whatever out of the question?

Virtually all user-facing functions are waiting for I/O, and thus prime examples of functions that should be asynchronous. Asynchronous functions were literally invented for this use case: Efficiently execute many tasks that need to wait for external events, like the arrival of an HTTP response.

Python's threads-which-fight-for-the-GIL can sugarcoat bad API design, but why would you force your users to do that? Having one event loop that executes multiple minio API calls asynchronously is the sane and efficient design.

**Comment by harshavardhana** [comment]
> Your dependencies' problems are your problems - is switching to aiohttp or whatever out of the question?

Whatever you want please send a PR and we can take a look or use what we have. 

This is not a priority for us.
